### DISCUSSION TITLE
Import from prior version?

### DISCUSSION THREAD

**Discussion by alogcat** [nextcloud/cookbook#2164] [FAQ and user questions] [answered]
I had cookbook set up, and had to reinstall Nextcloud without getting proper backups. I saw in the FAQ that you can adjust where the recipes should be found, which can reset the database, but I did that after copying all the prior recipe folders back into Recipes, and it still can't see them. I tested by creating a new recipe, and it creates in the same folder, so it's definitely able to use it. 

Can it be set up so that imports can be done from JSON files that are formatted from Cookbook? Or if that's an option, I don't see how to do it. Thanks for all your work, it's a great tool!

**Comment by seyfeb** [comment]
Hey! Try clicking the `Rescan Library` button as seen here in the "Where are my files" section

https://nextcloud.github.io/cookbook/user/ 

https://nextcloud.github.io/cookbook/user/assets/settings.png

**Reply by alogcat** [reply]
I did try that a number of times, and it never recognized that there were any files in that directory. It could create files there, so it had permissions to write at least.

**Comment by alogcat** [comment, marked as answer]
What I ended up doing was copying over the old Recipes folder to the admin account's Recipes, and then sharing it down to the user level. At some point in the cookbook admin, I was able to see the folders, but they had a size of -1, so I clicked on each subfolder and then back to Recipes, and it would scan and finally show the size, so that meant it had scanned it. Doing it manually wasn't ideal, but it seems to work. Rescan button worked after that. 
Then in each user, you'd have to go in and make sure it was pointed to the new shared Recipes folder, and rescan, and it would then load the recipes.

**Reply by seyfeb** [reply]
Sounds quite complicated, but I'm not sure what went wrong there.. I'm glad you were able to get your recipe back into the system!

**Reply by christianlupus** [reply]
I guess the problem was that the folder was copied on FS level, not on the NC server level. So, the files needed to be cached in `oc_files`.

**Comment by keenapps** [comment]
Hi Folks, since i had the exact same problem and was struggling to get over 900 recipes back in the cookbook i thought i leave you are more forward and simple solution.
In my case i reinstalled the server and switched to nextcloud (and cookbook) in a docker container.
As @christianlupus was guessing correctly the folder was copied on FS Level and not NC Server Level.
With this great hint i found a solution in the nextcloud documentation and discussions.

Long story simplified, the database is not aware of the files..

Solution:
`sudo docker exec -u www-data *ContainerID* php /var/www/html/occ files:scan --all -vvv` 

Without using docker it should be:
`sudo -u www-data php /var/www/html/occ files:scan --all -vvv`

This scans your nextcloud folder for new files and adds them to your database with a visual progress.
It may take a while depending on the amount of files and storage you use. 
Maybe depending on your usecase you may want to fine tune this to specific users and folders.

After this open your cookbook and scan your recipes folder again. After your reload the browser-tab / cookbook everything is back.
